tuplex/core/src/physical/ResolveTask.cc

```diff
@@ -816,113 +865,169 @@ namespace tuplex {
             _normalPtrBytesRemaining = _partitions[_currentNormalPartitionIdx]->bytesWritten();
             _normalRowNumber = 0;
             _rowNumber = 0;
+        } else {
+            _currentNormalPartitionIdx = 0;
+            _normalPtr = nullptr;
+            _normalPtrBytesRemaining = 0;
+            _normalNumRows = 0;
+            _normalRowNumber = 0;
+            _rowNumber = 0;
+        }
 
-            // merge exceptions with normal rows after calling slow code over them...
-            // basic idea is go over all exception partitions, execute row wise the resolution function
-            // and merge the result back to the partitions
-            for(auto partition : _exceptions) {
-                const uint8_t* ptr = partition->lockRaw();
-                int64_t numRows = *((int64_t*)ptr); ptr += sizeof(int64_t);
-
-                for(int i = 0; i < numRows; ++i) {
-
-                    // old
-                    // _currentRowNumber = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    // int64_t ecCode = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    // int64_t operatorID = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    // int64_t eSize = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-
-                    const uint8_t *ebuf = nullptr;
-                    int64_t ecCode = -1, operatorID = -1;
-                    size_t eSize = 0;
-                    auto delta = deserializeExceptionFromMemory(ptr, &ecCode, &operatorID, &_currentRowNumber, &ebuf,
-                                                                &eSize);
+        size_t runInd;
+        int64_t runRemaining;
+        const uint8_t *runPtr = nullptr;
+        if (_runtimeExceptions.size() > 0) {
+            runInd = 0;
+            runPtr = _runtimeExceptions[runInd]->lockRaw();
+            runRemaining = *((int64_t *) runPtr); runPtr += sizeof(int64_t);
+        }
 
-                    processExceptionRow(ecCode, operatorID, ebuf, eSize);
-                    ptr += delta;
+        size_t inputInd;
+        int64_t inputRemaining;
+        int64_t inputTotal = _numInputExceptions;
+        const uint8_t *inputPtr = nullptr;
+        if (_numInputExceptions > 0) {
+            inputInd = _inputExceptionIndex;
+            inputPtr = _inputExceptions[inputInd]->lockRaw();
+            inputRemaining = *((int64_t *) inputPtr) - _inputExceptionOffset; inputPtr += sizeof(int64_t);
+            for (int i = 0; i < _inputExceptionOffset; ++i) {
+                int64_t* ib = (int64_t*)inputPtr;
+                auto eSize = ib[3];
+                inputPtr += eSize + 4*sizeof(int64_t);
+            }
+        }
 
-                    // logger.debug("processing potential output row #" + std::to_string(_currentRowNumber));
+        size_t inputProcessed = 0;
+        const uint8_t **ptr;
```

Comment:
always initialize variables, i.e. ptr = nullptr; Also this is an array of ptrs. Maybe rename to ptr_array?